common/src/main/java/org/apache/falcon/entity/CatalogStorage.java

```diff
@@ -100,12 +100,20 @@ protected CatalogStorage(String catalogUrl, String tableUri) throws URISyntaxExc
         if (catalogUrl == null || catalogUrl.length() == 0) {
             throw new IllegalArgumentException("Catalog Registry URL cannot be null or empty");
         }
-
+        verifyAndUpdateConfiguration(getConf());
         this.catalogUrl = catalogUrl;
 
         parseFeedUri(tableUri);
     }
 
+    private void verifyAndUpdateConfiguration(Configuration conf) {
```

Comment:
I assume you are always using getConf() as the input argument? Then, a better way is to have verifyAndUpdateConfiguration() with no input and call getConf() in the function, so you don't need to reset configuration with the same value.